internal/api/impl/auth/auth.go

```diff
@@ -28,6 +30,19 @@ import (
 	"github.com/perses/perses/pkg/model/api/config"
 )
 
+func getRedirectURI(r *http.Request, authKind string, slugID string) string {
+	rd := url.URL{}
+	rd.Host = util.GetRequestHost(r)
```

Comment:
I don't think this will work the way you would like. This method (`GetRequestHost`) is calling another method from another package: `middlewareapi.GetRequestScope(req)`: https://github.com/oauth2-proxy/oauth2-proxy/blob/master/pkg/apis/middleware/scope.go

It basically extracts the struct `RequestScope` from the request context. But it's only possible if this value is set before (as stated by the struct comment, probably set by a middleware to share information from a middleware to another one) and it's not the case in our current implementation.